pkg/migrate/cstor/pool.go

```diff
@@ -477,24 +472,41 @@ func getCSP(cspLabel string) (*apis.CStorPool, error) {
 
 // The old pool pod should be scaled down before the new cspi pod reconcile is
 // enabled to avoid importing the pool at two places at the same time.
-func (c *CSPCMigrator) scaleDownDeployment(cspObj *apis.CStorPool, openebsNamespace string) error {
-	klog.Infof("Scaling down deployment %s", cspObj.Name)
+func (c *CSPCMigrator) scaleDownDeployment(cspName, cspiName, openebsNamespace string) error {
+	var zero int32 = 0
+	klog.Infof("Scaling down csp deployment %s", cspName)
 	cspDeployList, err := c.KubeClientset.AppsV1().
 		Deployments(openebsNamespace).List(
 		metav1.ListOptions{
-			LabelSelector: "openebs.io/cstor-pool=" + cspObj.Name,
+			LabelSelector: "openebs.io/cstor-pool=" + cspName,
 		})
 	if err != nil {
 		return err
 	}
 	if len(cspDeployList.Items) != 1 {
-		return errors.Errorf("invalid number of csp deployment found for %s: expected 1, got %d", cspObj.Name, len(cspDeployList.Items))
+		return errors.Errorf("invalid number of csp deployment found for %s: expected 1, got %d", cspName, len(cspDeployList.Items))
+	}
+	newCSPDeploy := cspDeployList.Items[0]
+	cspiDeploy, err := c.KubeClientset.AppsV1().
+		Deployments(openebsNamespace).Get(cspiName, metav1.GetOptions{})
+	if err != nil {
+		return errors.Wrapf(err, "failed to get deployment for cspi %s", cspiName)
+	}
+	// While scaling down the csp deployment changing the
+	// volumes as well so that the zrepl.lock file used by
+	// csp and cspi becomes the same to avoid data corruption
+	// due to multiple imports at the same time.
+	newCSPDeploy.Spec.Replicas = &zero
+	newCSPDeploy.Spec.Template.Spec.Volumes = cspiDeploy.Spec.Template.Spec.Volumes
+	patchData, err := GetPatchData(cspDeployList.Items[0], newCSPDeploy)
+	if err != nil {
+		return errors.Wrapf(err, "failed to patch data for csp %s", cspName)
```

Comment:
nit: 
```suggestion
		return errors.Wrapf(err, "failed to get patch data for csp deployment %s", cspName)
```

Reply:
The error returned by GetPatchData function is wrapped like "failed to generate patch data"